### ISSUE TITLE
Redirecting Create on Create doesn't clear values

### ISSUE THREAD

**Issue by fab1an** [marmelab/admin-on-rest#941]
I would like to Create on Enter press and, after success redirect to an empty Create again.

I've tried everything described here: https://marmelab.com/admin-on-rest/CreateEdit.html#redirection-after-submission but it doesn't seem to be possible.

I get messages about history not being able to push the same route twice. 

Is there any way to do this?

**Comment by djhi** [comment]
Reproduced. Indeed, the form values aren't cleared when `redirect` is `false`. I'm not sure about considering it a bug though. @fzaninotto, your opinion on this ?

In the mean time, you can probably achieve that with a custom [saga](https://marmelab.com/admin-on-rest/Admin.html#customsagas):

```js
// In resetCreateFormSaga.js
import { takeEvery, put } from 'redux-saga/effects'
import { reset } from 'redux-form';
import { CRUD_CREATE_SUCCESS } from 'admin-on-rest';

export function* resetCreateForm(action) {
    // you may trigger this reset only if the action targets a specific resource
    if (action.meta.resource === 'MyResource') {
        yield put(reset('record-form'));
    }
}

export default function* resetCreateFormSaga() {
    yield takeEvery(CRUD_CREATE_SUCCESS, resetCreateForm);
}
```

I'll keep this issue open until we decide whether or not it should be considered a bug.

**Comment by fab1an** [comment]
@djhi Where does that `'record-form'` identifier come from to pass to redux-form?

**Comment by djhi** [comment]
That's the name name we use for all the forms in Admin-on-rest. It's hard coded

**Comment by djhi** [comment]
More info: It's being used in [SimpleForm](https://github.com/marmelab/admin-on-rest/blob/master/src/mui/form/SimpleForm.js#L67) and  [TabbedForm](https://github.com/marmelab/admin-on-rest/blob/master/src/mui/form/TabbedForm.js#L100)

**Comment by fab1an** [comment]
Thanks.

**Comment by fab1an** [comment]
Just discovered something interesting. If you wrap SimpleForm in another component and pass the props, the form gets cleared after submit and works like before:

```jsx
class FormWrapper extends React.Component {
    render() {
        return (
            <SimpleForm redirect="false" {...this.props}>
                <TextInput source="field" />
            </SimpleForm>
        )
    }
}

const render = (props) => 
    <Create {...props}>
         <FormWrapper/>
    </Create>
```

Any idea what's going on?

**Comment by djhi** [comment]
FYI I can't reproduce the problem on the `next` branch

**Comment by fzaninotto** [comment]
Well then, we'll consider it fixed by `next`.

**Comment by fab1an** [comment]
Very good!

**Comment by djhi** [comment]
Should have been fixed by `1.3.0`. @fab1an can you confirm ?
